src/analyzer/Analyzer.cc

```diff
@@ -470,25 +466,33 @@ bool Analyzer::Remove()
 
 void Analyzer::PreventChildren(zeek::Tag tag)
 	{
-	auto it = std::find(prevented.begin(), prevented.end(), tag);
-
-	if ( it != prevented.end() )
+	if ( IsPreventedChildAnalyzer(tag) )
 		return;
 
 	prevented.emplace_back(tag);
 	}
 
+bool Analyzer::IsPreventedChildAnalyzer(const zeek::Tag& tag) const
+	{
+	return std::find(prevented.begin(), prevented.end(), tag) != prevented.end();
+	}
+
 bool Analyzer::HasChildAnalyzer(zeek::Tag tag)
+	{
+	return GetChildAnalyzer(tag) != nullptr;
+	}
```

Comment:
Since you are already touching this function, WDYT about fixing the parameter copy (not in line with other functions here), and fixing const correctness (but that's a bigger issue in this file)?

```suggestion
bool Analyzer::HasChildAnalyzer(const zeek::Tag& tag) const
	{
	return GetChildAnalyzer(tag) != nullptr;
	}
```

Reply:
I'm happy to do that (and hat it previously) but was only 99% sure about that being API compatible. So if you give a thumbs-up on that I'll do that :-)

Reply:
I did that, it seems it shouldn't matter for a normal user.